### PR TITLE
Clean up deps and move to v0.22.5

### PR DESCRIPTION
**Pull request by skonto**
As per title. Moves to v0.22.5 and tries to simply deps.

### CODE REVIEW

**Comment by skonto** [comment]
```
...
  level=warning msg="[runner] Can't run linter goanalysis_metalinter: buildir: failed to load package continual: could not load export data: no export data for \"knative.dev/eventing-kafka-broker/test/upgrade/continual\""
  level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: buildir: failed to load package continual: could not load export data: no export data for \"knative.dev/eventing-kafka-broker/test/upgrade/continual\"\n\n"
  
```

**Comment by skonto** [comment]
Kafka broker does not build.

**Comment by skonto** [comment]
@pierDipi seems Kafka post install fails.

File: go.mod [hunk 1]
```diff
@@ -22,13 +22,13 @@ require (
 	k8s.io/api v0.22.5
 	k8s.io/apimachinery v0.22.5
 	k8s.io/client-go v0.22.5
-	knative.dev/eventing v0.29.0
-	knative.dev/eventing-kafka v0.27.0
+	knative.dev/eventing v0.30.1
+	knative.dev/eventing-kafka v0.28.0
```
Comment (matzew) [inline go.mod:26]: why 0.30 VS 0.28 ?

**Comment by matzew** [comment]
On 

```
FLOATING_DEPS=(
  "knative.dev/hack@v0.0.0-20220224013837-e1785985d364"
  "knative.dev/networking@${KN_VERSION}"
  "knative.dev/operator@${KN_VERSION}"
  "knative.dev/pkg@v0.0.0-20220223180940-4fcbc1bc12e8"
)
```

I think for `hack` and `pkg` we should also _align_ with the `KN_VERSION`?

**Comment by matzew** [comment]
FWIW: this bumps some deps ETC for all of eventing components:
* https://github.com/openshift-knative/serverless-operator/pull/1489/files

Perhaps it makes sense to place this PR ontop of that ? :shrug:

**Comment by skonto** [comment]
> I think for hack and pkg we should also align with the KN_VERSION?

I tried to use release-1.2 but it fails, the branches are not aligned. Mentioned it on slack [here](https://coreos.slack.com/archives/CD87JDUB0/p1649929839668189?thread_ts=1649921950.588079&cid=CD87JDUB0). 
`go mod` forces that.  If you have a specific combination that works let me know.

> Perhaps it makes sense to place this PR ontop of that ? 

We can do that too wfm.

File: go.mod (hunk 1 above)
Comment (skonto) [reply go.mod:26]: I don't set it manually, `hack/update-deps.sh --upgrade` does the update.

**Comment by skonto** [comment]
closi

**Comment by skonto** [comment]
Closing this in favor of https://github.com/openshift-knative/serverless-operator/pull/1515 etc

**Pull request closed without merging**